### ISSUE TITLE
ScyllaCluster Pods not coming up when deploying multiple Racks

### ISSUE THREAD

**Issue by rzetelskik** [scylladb/scylla-operator#755]
**Describe the bug**
When trying to deploy a ScyllaCluster with multiple Racks, none of the Pods will come up.

**To Reproduce**
1. Deploy Scylla Operator and Scylla Manager
2. Deploy a ScyllaCluster (provided in artifacts) with >1 Racks

**Expected behavior**
All Pods should come up.

**Config Files**
[artifacts.tar.gz](https://github.com/scylladb/scylla-operator/files/6976058/artifacts.tar.gz)

**Logs**
```
ERROR 2021-08-12 13:19:14,191 [shard 0] rpc - client 10.102.116.70:7000: fail to connect: Connection refused
...
ERROR 2021-08-12 13:20:04,702 [shard 0] token_metadata - sorted_tokens is empty in first_token_index!
...
ERROR 2021-08-12 13:20:04,750 [shard 0] init - Startup failed: std::runtime_error (sorted_tokens is empty in first_token_index!)
2021-08-12 13:20:04,773 INFO exited: scylla (exit status 1; not expected)
```

Log dump provided in artifacts.

**Environment:**
 - Platform: minikube
 - Kubernetes version: 1.21
 - Scylla version: 4.4.0
 - Scylla-operator version: master

**Additional context**
@zimnx confirmed he has the same issue and suggested this could be due to starting all Racks at once.

**Comment by rzetelskik** [comment]
I've also managed to reproduce this behaviour on GKE.
Scylla logs: [gke.tar.gz](https://github.com/scylladb/scylla-operator/files/6976790/gke.tar.gz)

**Comment by tnozicka** [comment]
we need e2e coverage

**Comment by rzetelskik** [comment]
I'll open a separate issue for adding multi-rack e2e then

**Comment by tnozicka** [comment]
>I'll open a separate issue for adding multi-rack e2e then

Wouldn't you add it in the same PR? If it's broken now, the fix should prove it actually fixes it by adding and e2e, and you can't merge the test before the fix because it should fail.

**Comment by tnozicka** [comment]
@dkropachev is QA covering this case?

**Comment by dkropachev** [comment]
> @dkropachev is QA covering this case?

We recently added test case to functional suite that does the following:
1. Deploy ScyllCluster with one rack
2. Addes two nodes, one by one to another rack

No exactly the same, but close.

**Comment by rzetelskik** [comment]
fixed with #843
